Apps/WebClient/src/ClientApp/src/models/errors.ts

```diff
@@ -28,6 +33,31 @@ export interface ResultError {
     actionCode?: ActionType;
     // The HTTP status code returned by the request.
     statusCode?: number;
+
+    constructor(
+        errorCode: string,
+        resultMessage: string,
+        traceId: string = "",
+        statusCode?: number
+    ) {
+        super(resultMessage);
+        this.errorCode = errorCode;
+        this.traceId = traceId;
+        this.statusCode = statusCode;
+    }
+
+    public static fromApiResultError(apiResultError: ResultError): ResultError {
```

Comment:
It's a little confusing to have apiResultError typed as `ResultError`.  I'd suggest introducing an `ApiResultError` interface and referencing that interface in the `RequestResult` model in place of `ResultError`.

We could also consider reserving the "Error" name for our JS Error classes and giving the back-end error models (and banner error models) a different moniker.

Reply:
I like this suggestion, I'll do that it'll be much cleaner 👍 Not sure what other moniker would fit but I'll think of some synonyms or the likes.

Reply:
![image](https://github.com/bcgov/healthgateway/assets/19548348/a178b703-d49c-4d7f-a70a-4aef2de032ec)

I like `ResultBooBoo`

Reply:
You could also append a word, so you'd get something like ErrorDetails.